### PR TITLE
Improve rigidity slider handling

### LINKED ISSUE #1496: Combine subsequent rigidity changes in the editor to a single action
to make undoing easier

### LINKED ISSUE #1533: Inactive slider doesn't detect mouse up
might be a Godot bug:
- Use all MP, try to move the rigidity bar
- Keep mouse held down while moving the mouse off the slider to the undo button
- Undo changes
- Move cursor back to the rigidity bar and see the slider value jump to the mouse without pressing anything

### PR DESCRIPTION
**Pull request by 84634E1A607A**
**Brief Description of What This PR Does**

Disabling the slide prevents sliding back and makes it a must to use undo. And if the player added something else after that, he can only undo all those things before he can use the slide bar again. So I just deleted that section of code.

Rigidity changes are combined.

**Related Issues**

Fixes #1496

**Progress Checklist**

Note: before starting this checklist the issue should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [ ] Initial code review passed
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author.

### CODE REVIEW

File: src/microbe_stage/editor/MicrobeEditorAction.cs
Comment (FuJa0815) [inline src/microbe_stage/editor/MicrobeEditorAction.cs:28]: Could you restore the old order of those variables? This diff is very hard to read, at least for me.
And wouldn't properties with private setters be better here?

**Review by FuJa0815 — CHANGES_REQUESTED** [review]

File: src/general/ActionHistory.cs
Comment (hhyyrylainen) [inline src/general/ActionHistory.cs:48]: Typo?

File: src/microbe_stage/editor/MicrobeEditor.cs
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:78]: I really dislike doing a special case for initialRigidity. If this MP compare to initial rigidity is wanted, the full dynamic recompute approach should be done at once. There's an abandoned PR about that: https://github.com/Revolutionary-Games/Thrive/pull/2365

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

File: src/microbe_stage/editor/MicrobeEditorAction.cs
Comment (84634E1A607A) [reply src/microbe_stage/editor/MicrobeEditorAction.cs:28]: Rider suggests readonly variables be placed before non-readonly ones. The change here is simply changing the modifiers from private to public.

Comment (84634E1A607A) [reply src/microbe_stage/editor/MicrobeEditorAction.cs:28]: They are readonly ones. Can't be set outside constructor.

File: src/microbe_stage/editor/MicrobeEditor.cs
Comment (84634E1A607A) [reply src/microbe_stage/editor/MicrobeEditor.cs:78]: I know. But it is abandoned; and I cannot think of better solutions.

Comment (84634E1A607A) [reply src/microbe_stage/editor/MicrobeEditor.cs:78]: I'll delete this then.

File: src/microbe_stage/editor/MicrobeEditorAction.cs
Comment (hhyyrylainen) [reply src/microbe_stage/editor/MicrobeEditorAction.cs:28]: Public *fields* should not be used. Use properties if needed, but I think there might be a better solution out there rather than doing this. I would rather add like a method to a rigidity change action class that allows modifying it, and then using a cast / type check to try to cast the latest action to that type to access the method.

File: src/microbe_stage/editor/MicrobeEditor.cs
Comment (hhyyrylainen) [reply src/microbe_stage/editor/MicrobeEditor.cs]: I was suggesting that if you are willing to pick it up, you could continue that PR.

Comment (84634E1A607A) [reply src/microbe_stage/editor/MicrobeEditor.cs]: Well, tried, failed. Too many problems there. Some out of my talent...

Comment (hhyyrylainen) [reply src/microbe_stage/editor/MicrobeEditor.cs]: Maybe someone else on the team could help with that? From what I remember that PR wasn't far off from being finished, just a little bit of refactoring to separate the cost and actual applying of an action functions.

Comment (84634E1A607A) [reply src/microbe_stage/editor/MicrobeEditor.cs]: It failed to handle overlaps (place sth on a cytoplasm costs a place *and* a delete; and if I place another cytoplasm elsewhere then it costs a move. I find myself unable to handle that. Maybe I should think harder.)

File: src/microbe_stage/editor/MicrobeEditorAction.cs
Comment (84634E1A607A) [reply src/microbe_stage/editor/MicrobeEditorAction.cs]: Okay, but I'm not fully sure what I've done is what you've said.

Comment (hhyyrylainen) [reply src/microbe_stage/editor/MicrobeEditorAction.cs]: These are *fields* not *properties* there is a big difference between those in C#.

Comment (84634E1A607A) [reply src/microbe_stage/editor/MicrobeEditorAction.cs]: Actually I'm wondering if I've got *"add like a method to a rigidity change action class"* right. However, thanks for telling me not to use public fields.

**Comment by 84634E1A607A** [comment]
This PR conflicts with #2445. I'll wait until that one is merged.

**Pull request closed without merging**
